### ISSUE TITLE
Setting markeredgecolor raises a warning

### ISSUE THREAD

**Issue by u55** [matplotlib/matplotlib#6480]
Hi matplotlib developers,

Setting `markeredgecolor` to an RGB or RGBA numpy array now gives a warning in matplotlib 1.5.1.  This was not a problem in version 1.4.3.  Here is a minimal working example:

``` python
import numpy as np
import matplotlib.pyplot as plt

# color = [0.5, 0.5, 0.5, 1.0]  # Works without a warning
color = np.array([0.5, 0.5, 0.5, 1.0])  # Gives a warning

plt.figure()
plt.plot(np.arange(9), np.arange(9), marker='o', markeredgecolor=color)
plt.show()
```

In python 2.7, it gives this warning:

```
/usr/lib/python2.7/site-packages/matplotlib/lines.py:1082: UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if self._markeredgecolor != ec:
```

And in python 3.5, it gives this different warning:

```
/usr/lib/python3.5/site-packages/matplotlib/lines.py:1082: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if self._markeredgecolor != ec:
```

Tested on Arch Linux.
--Thanks.

**Comment by u55** [comment]
@tacaswell Git is blaming you for [this line](https://github.com/matplotlib/matplotlib/blame/6f27d8b340237afd102dd50c05603dac4893036e/lib/matplotlib/lines.py#L1147).

**Comment by tacaswell** [comment]
Those are both saying that we are comparing a string(Unicode) and an array.

I have a caught memory that we fixed this on master.

The change might have actually been updating Python or numpy underneath.

**Comment by jenshnielsen** [comment]
@tacaswell We have fixed similar issues but I don't think we have fixed this one.

**Comment by u55** [comment]
Yes, there is a related open issue here #5806 and a closed issue here #5209.

**Comment by u55** [comment]
And issue #5209 was fixed with this commit 5e8fb62.

**Comment by dstansby** [comment]
I'm getting something slightly different on python 3.5:

```
>>> plt.plot(np.arange(9), np.arange(9), marker='o', markeredgecolor=color)
/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/lines.py:1160: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
```

Just running tests on a fix.
